core/src/main/scala/com/evolutiongaming/catshelper/Log.scala

```diff
@@ -44,30 +44,66 @@ object Log {
   object Mdc {
 
     private object Empty extends Mdc
-    private final case class Context(values: NonEmptyMap[String, String]) extends Mdc {
-      override def toString: String = s"MDC(${values.toSortedMap.mkString(", ")})"
-    }
+    private final case class Context(getValues: () => Map[String, String]) extends Mdc {
```

Comment:
Might be better add new lazy implementation and keep eager one as is?

Reply:
When Mdc.fromMap/fromList is called, the parameters will be evaluated anyway, so it's actually eager (but implemented a bit differently)

Reply:
additional reason of using eager implementation in eager factories is in avoiding `lazy val` where its definitely not needed (due to already pre-evaluated mdc map)

Reply:
Previous comment dropped by mistake:

> Then please elaborate why eager implementation should be done on top of lazy? I see no reason for it and prefer to keep both of them, but separately making the logic strait-forward and explicit

Reply:
Discussed with @t3hnar and agreed to base lazy implementation on the eager one